hail/src/main/scala/is/hail/expr/ir/agg/NDArraySumAggregator.scala

```diff
@@ -0,0 +1,132 @@
+package is.hail.expr.ir.agg
+
+import is.hail.annotations.{Region, StagedRegionValueBuilder}
+import is.hail.asm4s._
+import is.hail.expr.ir.functions.StringFunctions
+import is.hail.expr.ir.{EmitCode, EmitCodeBuilder, EmitRegion, coerce}
+import is.hail.types.physical.{PBooleanRequired, PCanonicalTuple, PCode, PNDArray, PNDArrayCode, PNDArrayValue, PNumeric, PType}
+import is.hail.utils._
+
+class NDArraySumAggregator (ndTyp: PNDArray) extends StagedAggregator {
+  override type State = TypedRegionBackedAggState
+
+  // State needs:
+  // 1. Initialized or not
+  // 2. The ndarray itself.
+
+  override def resultType: PType = ndTyp
+
+  val stateType = PCanonicalTuple(true, ndTyp)
+
+  override def initOpTypes: Seq[PType] = Array[PType]()
+
+  override def seqOpTypes: Seq[PType] = Array(ndTyp)
+
+  val ndarrayFieldNumber = 0
+
+  def isInitialized(state: State): Code[Boolean] = {
+    stateType.isFieldDefined(state.off, ndarrayFieldNumber)
+  }
```

Comment:
Now that you're just using the missing bit, you shouldn't need this or `ndArrayPointer`. Instead of
```
cb.ifx(isInitialized(state), {
  val currentNDPValue = PCode(ndTyp, ndArrayPointer(state)).asNDArray.memoize(cb, "ndarray_sum_seqop_current")
  ...
}, {
  // uninitialized case
})
```
you can simplify to
```
val statePV = new PCanonicalBaseStructSettable(stateType, state.off)
...
statePV.loadField(ndarrayFieldNumber).consume(cb, {
  // uninitialized case
}, { currentNDPCode =>
  val currentNDPValue = currentNDPCode.asNDArray.memoize(cb, "ndarray_sum_seqop_current")
  ...
})
```
A bit simpler and more idiomatic.